whisperv6/whisper.go

```diff
@@ -914,6 +938,14 @@ func (whisper *Whisper) runMessageLoop(p *Peer, rw p2p.MsgReadWriter) error {
 		}
 
 		packet.Discard()
+
+		if packet.Code != p2pMessageCode && whisper.ratelimiter != nil {
```

Comment:
Maybe move this code at the beginning of the loop (before `switch` statement on type): there are some returns along the code, which means this ratelimiter code will not be executed when invalid packets are arrived. This theoretically opens up a way to bypass ratelimiting by spamming with incorrect envelopes.(this is probably not a problem, because peers will be disconnected, but maybe there is a valid point anyway)

also, can be extracted to separate method with a clear naming probably.

Reply:
i didn't want to waste (discard without passing to a user) already received traffic, as it is not clear if last portion is spam or not. but right i will keep removing after packet was processed and move accounting to the top